tesseract_kinematics/src/opw/opw_inv_kin.cpp

```diff
@@ -66,7 +66,8 @@ IKSolutions OPWInvKin::calcInvKin(const Eigen::Isometry3d& pose,
       harmonizeTowardZero<double>(eigen_sol);  // Modifies 'sol' in place
 
       // Add solution
-      solution_set.push_back(eigen_sol);
+      if (tesseract_common::satisfiesPositionLimits(eigen_sol, limits_.joint_limits))
```

Comment:
Off topic, but I notice at 58 we aren't reserving any room for redundant solutions.

Reply:
I am not sure there is a good way to calculate the number of redundant solutions before hand given the position, though it should not be difficult to calculate the max number of redundant solutions based on the joint limits.